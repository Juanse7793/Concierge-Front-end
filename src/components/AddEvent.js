import React, { useState } from 'react';
import TextField from '@mui/material/TextField';
// import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
// import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
// import { DatePicker } from '@mui/x-date-pickers/DatePicker';
import { useDispatch } from 'react-redux';
import { addEvent } from '../redux/reducers/events';

function AddEvent() {
  // eslint-disable-next-line camelcase
  const [start_date, setStart] = useState();
  // eslint-disable-next-line camelcase
  const [end_date, setEnd] = useState();
  const [name, setName] = useState();
  const [location, setLocation] = useState();
  const [price, setPrice] = useState();

  const dispatch = useDispatch();

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(addEvent({
      start_date, end_date, name, location, price,
    }));
    setName('');
    setLocation('');
    setPrice('');
    setStart('');
    setEnd('');
  };

  const sx = {
    width: '500px',
    color: 'black',
  };

  return (
    <div className="add-event-main">
      <div className="add-event-header">
        <h1 className="add-event-title">Add Event</h1>
      </div>
      <form className="add-event-form" onSubmit={handleSubmit}>
        <TextField
          value={name}
          onChange={(e) => setName(e.target.value)}
          sx={sx}
          id="event-name-input"
          label="Event Name"
          margin="normal"
        />
        <TextField
          value={location}
          onChange={(e) => setLocation(e.target.value)}
          sx={sx}
          id="event-location-input"
          label="Event Location"
          margin="normal"
        />
        <TextField
          value={price}
          onChange={(e) => setPrice(e.target.value)}
          sx={sx}
          id="event-price-input"
          label="Event Price"
          margin="normal"
        />
        <div className="add-event-dates">
          <h2 className="add-event-dates-title">Event dates</h2>
        </div>
        <TextField
          // eslint-disable-next-line camelcase
          value={start_date}
          onChange={(e) => setStart(e.target.value)}
          sx={sx}
          id="event-start-input"
          label="Event Start"
          margin="normal"
        />
        <TextField
          // eslint-disable-next-line camelcase
          value={end_date}
          onChange={(e) => setEnd(e.target.value)}
          sx={sx}
          id="event-end-input"
          label="Event End"
          margin="normal"
        />
        <div className="add-event-images">
          <h2 className="add-event-images-title">Event images</h2>
        </div>
        <input type="file" multiple id="new-event-image" />
        <input type="submit" id="submit-event" value="Add Event" />
      </form>
    </div>
  );
}

export default AddEvent;
