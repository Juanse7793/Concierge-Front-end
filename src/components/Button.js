import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
  disabled, text, func, className,
}) => (
  <button
    type="button"
    className={`pill ${className}`}
    disabled={disabled}
    onClick={() => func()}
  >
    {text}
  </button>
);

Button.defaultProps = {
  className: '',
};

Button.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
  func: PropTypes.func.isRequired,
};

export default Button;
