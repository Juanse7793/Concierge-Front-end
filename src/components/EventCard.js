import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const EventCard = ({ event }) => (
  <Link to={`/events/${event.id}`} className="event-card column">
    <div className="circle">
      <img src={event.image} className="eventImage" alt="" />
    </div>
    <h3 className="event-card-title">{event.name}</h3>
    <p className="event-card-details">{event.location}</p>
    <p className="event-card-details">
      $
      {event.price}
    </p>
  </Link>
);

EventCard.propTypes = {
  event: PropTypes.objectOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      image: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
    }),
  ).isRequired,
};

export default EventCard;
