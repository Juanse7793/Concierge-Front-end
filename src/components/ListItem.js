import React from 'react';
import PropTypes from 'prop-types';

const ListItem = ({
  name, start, end, deleteFunc,
}) => (
  <li className="row pill list">
    <p>{name}</p>
    <div className="row">
      <small>{`${start} ~ ${end}`}</small>
      <button type="button" className="red pill link" onClick={deleteFunc}>Delete</button>
    </div>
  </li>
);

ListItem.propTypes = {
  name: PropTypes.string.isRequired,
  start: PropTypes.string.isRequired,
  end: PropTypes.string.isRequired,
  deleteFunc: PropTypes.func.isRequired,
};

export default ListItem;
