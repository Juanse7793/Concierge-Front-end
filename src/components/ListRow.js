import React from 'react';
import PropTypes from 'prop-types';

const ListRow = ({ caption, text }) => (
  <li className="row">
    <span>{caption}</span>
    <span>{text}</span>
  </li>
);

ListRow.propTypes = {
  caption: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default ListRow;
