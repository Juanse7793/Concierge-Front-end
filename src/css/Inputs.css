.out {
  position: relative;
  margin: 1rem;
  margin-top: 0;
}

.out input {
  border: 2px solid white;
  margin: 0;
  width: 315px;
}

.out label {
  position: absolute;
  font-size: 1rem;
  font-weight: bold;
  font-family: Tahoma;
  text-transform: uppercase;
  top: calc(1rem + 2px);
  left: var(--left);
  transition: all 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.out.focus label {
  cursor: pointer;
  font-size: small;
  font-family: var(--heading-font);
  left: calc(var(--left) - 1rem);
  top: -0.1rem;
  background: var(--theme-lime);
  line-height: 0.3;
  padding: 0 0.2rem;
}

.react-datepicker__day--in-range,
.react-datepicker__day--in-selecting-range {
  background-color: var(--theme-lime) !important;
}

.react-datepicker__day--keyboard-selected {
  background-color: #b6e222 !important;
}
