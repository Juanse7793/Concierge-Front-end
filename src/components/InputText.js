import React from 'react';
import PropTypes from 'prop-types';
import '../css/InputText.css';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

export const InputText = ({
  text,
  value,
  func,
  required,
  className,
  autoComplete,
}) => {
  const focus = (e) => {
    console.log('focus', e, e.parentNode);
    e.parentNode.classList.toggle('focus', e === document.activeElement || e.value !== '');
  };

  return (
    <div className={`out ${className}`}>
      <input
        type="text"
        name={text.toLowerCase()}
        value={value}
        onChange={(e) => func(e)}
        required={required}
        onFocus={(e) => focus(e.target)}
        onBlur={(e) => focus(e.target)}
        autoComplete={autoComplete}
        className="pill green white-border"
      />
      <label htmlFor={text.toLowerCase()}>{text}</label>
    </div>
  );
};

InputText.defaultProps = {
  className: '',
  required: false,
  autoComplete: 'off',
};

InputText.propTypes = {
  className: PropTypes.string,
  required: PropTypes.bool,
  autoComplete: PropTypes.string,
  text: PropTypes.string.isRequired,
  func: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export const DateRange = ({
  minDate,
  maxDate,
  startDate,
  endDate,
  func,
}) => (
  <div className="out focus">
    <DatePicker
      selectsRange
      minDate={minDate}
      maxDate={maxDate}
      startDate={startDate}
      endDate={endDate}
      popperPlacement="top-end"
      onChange={(e) => func(e)}
      className="pill green white-border"
    />
    {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
    <label>Reserved Dates</label>
  </div>
);

DateRange.defaultProps = {
  minDate: new Date('2000/12/12'),
  maxDate: new Date('2100/12/12'),
};

DateRange.propTypes = {
  minDate: PropTypes.instanceOf(Date),
  maxDate: PropTypes.instanceOf(Date),
  startDate: PropTypes.instanceOf(Date).isRequired,
  endDate: PropTypes.instanceOf(Date).isRequired,
  func: PropTypes.func.isRequired,
};
