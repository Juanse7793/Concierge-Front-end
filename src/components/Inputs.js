import React from 'react';
import PropTypes from 'prop-types';
import '../css/Inputs.css';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

export const InputText = ({
  text,
  value,
  func,
  required,
  className,
  autoComplete,
}) => {
  const focus = (e) => {
    e.parentNode.classList.toggle('focus', e === document.activeElement || e.value !== '');
  };

  return (
    <div className={`out ${className}`}>
      <input
        type="text"
        name={text.toLowerCase()}
        id={text.toLowerCase()}
        value={value}
        onChange={func}
        required={required}
        onFocus={(e) => focus(e.target)}
        onBlur={(e) => focus(e.target)}
        autoComplete={autoComplete}
        className="pill green white-border"
      />
      <label htmlFor={text.toLowerCase()}>{text}</label>
    </div>
  );
};

InputText.defaultProps = {
  className: '',
  required: false,
  autoComplete: 'off',
  value: '',
};

InputText.propTypes = {
  className: PropTypes.string,
  required: PropTypes.bool,
  autoComplete: PropTypes.string,
  text: PropTypes.string.isRequired,
  func: PropTypes.func.isRequired,
  value: PropTypes.string,
};

export const DateRange = ({
  text,
  minDate,
  maxDate,
  startDate,
  endDate,
  func,
}) => (
  <div className="out focus">
    <DatePicker
      selectsRange
      minDate={minDate}
      maxDate={maxDate}
      startDate={startDate}
      endDate={endDate}
      dateFormat="yyyy-MM-dd"
      popperPlacement="top-end"
      showDisabledMonthNavigation
      onChange={(e) => func(e)}
      className="pill green white-border"
    />
    {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
    <label>{text}</label>
  </div>
);

DateRange.defaultProps = {
  minDate: null,
  maxDate: null,
  startDate: null,
  endDate: null,
  text: 'Dates',
};

DateRange.propTypes = {
  text: PropTypes.string,
  minDate: PropTypes.instanceOf(Date),
  maxDate: PropTypes.instanceOf(Date),
  startDate: PropTypes.instanceOf(Date),
  endDate: PropTypes.instanceOf(Date),
  func: PropTypes.func.isRequired,
};
