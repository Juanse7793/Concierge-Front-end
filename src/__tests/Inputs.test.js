import renderer from 'react-test-renderer';
import { InputText, DateRange } from '../components/Inputs';

it('matches snapshot', () => {
  const component = renderer.create(
    <InputText
      text="name"
      value="Concierge"
      func={() => {}}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const component2 = renderer.create(
    <DateRange
      func={() => {}}
    />,
  );
  const tree2 = component2.toJSON();
  expect(tree2).toMatchSnapshot();
});
