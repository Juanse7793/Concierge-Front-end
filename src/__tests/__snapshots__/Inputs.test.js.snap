// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="out "
>
  <input
    autoComplete="off"
    className="pill green white-border"
    id="name"
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={false}
    type="text"
    value="Concierge"
  />
  <label
    htmlFor="name"
  >
    name
  </label>
</div>
`;

exports[`matches snapshot 2`] = `
<div
  className="out focus"
>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <input
        className="pill green white-border"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        readOnly={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <label>
    Dates
  </label>
</div>
`;
